grpc4bmi/bmi_julia_model.py

```diff
@@ -0,0 +1,615 @@
+        Perform all tasks that take place after exiting the model's time
+        loop. This typically includes deallocating memory, closing files and
+        printing reports.
+        """
+        self.implementation.finalize(self.state)
+
+    def get_component_name(self) -> str:
+        """Name of the component.
+        Returns
+        -------
+        str
+            The name of the component.
+        """
+        return self.implementation.get_component_name(self.state)
+
+    def get_input_item_count(self) -> int:
+        """Count of a model's input variables.
+        Returns
+        -------
+        int
+          The number of input variables.
+        """
+        return self.implementation.get_input_item_count(self.state)
+
+    def get_output_item_count(self) -> int:
+        """Count of a model's output variables.
+        Returns
+        -------
+        int
+          The number of output variables.
+        """
+        return self.implementation.get_output_item_count(self.state)
+
+    def get_input_var_names(self) -> List[str]:
+        """List of a model's input variables.
+        Input variable names must be CSDMS Standard Names, also known
+        as *long variable names*.
+        Returns
+        -------
+        list of str
+            The input variables for the model.
+        Notes
+        -----
+        Standard Names enable the CSDMS framework to determine whether
+        an input variable in one model is equivalent to, or compatible
+        with, an output variable in another model. This allows the
+        framework to automatically connect components.
+        Standard Names do not have to be used within the model.
+        """
+        return list(self.implementation.get_input_var_names(self.state))
+
+    def get_output_var_names(self) -> List[str]:
+        """List of a model's output variables.
+        Output variable names must be CSDMS Standard Names, also known
+        as *long variable names*.
+        Returns
+        -------
+        list of str
+            The output variables for the model.
+        """
+        return list(self.implementation.get_output_var_names(self.state))
+
+    def get_var_grid(self, name: str) -> int:
+        """Get grid identifier for the given variable.
+        Parameters
+        ----------
+        name : str
+            An input or output variable name, a CSDMS Standard Name.
+        Returns
+        -------
+        int
+          The grid identifier.
+        """
+        return self.implementation.get_var_grid(self.state, name)
+
+    def get_var_type(self, name: str) -> str:
+        """Get data type of the given variable.
+        Parameters
+        ----------
+        name : str
+            An input or output variable name, a CSDMS Standard Name.
+        Returns
+        -------
+        str
+            The Python variable type; e.g., ``str``, ``int``, ``float``.
+        """
+        return self.implementation.get_var_type(self.state, name)
```

Comment:
This will need a translation step. See https://docs.julialang.org/en/v1/manual/types/#Primitive-Types for an overview of Wflow types. It seems that just applying `.lower()` is sufficient (except for `char`...)

Reply:
added lower()